### PR TITLE
Private Events

### PR DESCRIPTION
**Pull request by GzimAsani**
**In this project:**

- I created the association between user, events and attendances
- The project was styled using bootstrap and customize css
- I  create sessions for user login and signup

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    if @user.valid? && @user.save
+      redirect_to login_path, notice: 'Account created successfuly, login using your credentials'
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @user = current_user
+  end
```
Comment (lekegitrepo) [inline app/controllers/users_controller.rb:18]: **_The UsersController `show` action method should_**

* [ ] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

* [ ] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

* [ ] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<h2 class = 'text-center mt-4'>Events History</h2>
+
+  <div class = 'container'>
+    <div class = 'row justify-content-md-center'>
+      <div class = 'col-8 mt-4'>
+        <div class="card text-center shadow mb-5 bg-info rounded">
+          <div>
+           <h4 class="p-2 w-50 mx-auto mb-2 text-center text-white"> Events you created: <%= current_user.events.count%></h4>
+          </div>
+          <% current_user.events.order('created_at DESC').each do |event|%>
```
Comment (lekegitrepo) [inline app/views/users/show.html.erb]: - There should be NO application logic in the views. Everything connected with data processing should be encapsulated in models and used in controllers.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                  <div class="card-footer bg-light text-dark px-2">
+                    <small class="">Date: [<%= event.date%>] </small>
+                  </div>
+                </div>
+              <%end%>
+           </div>
+          <%end%>
+
+          <h4 class="p-2 w-50 mx-auto mb-2 text-center text-white">Upcoming Events: <%= current_user.attended_events.where('date>?', Time.now).count%></h4>
+          <% current_user.attended_events.where('date>?', Time.now).each do |event|%>
```
Comment (lekegitrepo) [inline app/views/users/show.html.erb]: - There should be NO application logic in the views. Everything connected with data processing should be encapsulated in models and used in controllers.

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                  <div class="card-footer bg-light text-dark px-2">
+                      <small class="">Event will be held in <%= time_ago_in_words(event.date)%> [<%= event.date%>] </small>
+                  </div>
+                </div>
+              <%end%>
+           </div>
+          <%end%>
+
+          <h4 class="p-2 w-50 mx-auto mb-2 text-center text-white">Previously attended: <%= current_user.attended_events.where('date<?', Time.now).count%></h4>
+          <% current_user.attended_events.where('date<?', Time.now).each do |event|%>
```
Comment (lekegitrepo) [inline app/views/users/show.html.erb]: - There should be NO application logic in the views. Everything connected with data processing should be encapsulated in models and used in controllers.

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @GzimAsani :clap: :clap:
#### You have done a good job :+1: so far. The use of CSS shadow on some of the page elements is awesome. However, you need to implement the following required changes before this project can be approved.

#### User Interface:
- [Sign up page] There is a `password` label but the input placeholder says `Enter your email` and there is a `password_confirmation` label and the input placeholder says `Chose a password`
- [Login page] There is a `username` label but the input placeholder says `Enter your email`

#### UsersController:
- Update/Implement the show action to contain all the required events for the user. Check my inline comment below :point_down:

#### Logic in the view:
- _There should be NO application logic in the views. Everything connected with data processing should be encapsulated in models and used in controllers._
- _View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)_
- _There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. **You can check the following resources**_
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

#### Tests(failing tests):
- _Some(7) of the integration tests are failing._

#### _Reviewed by Temitayo Adeleke_

File: app/views/users/show.html.erb (hunk 4 above)
Comment (GzimAsani) [reply app/views/users/show.html.erb]: Thank you so much for your detailed review!

**Review by Temmarie — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @GzimAsani! 👋 👋 

#### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: :tada:

#### You may merge, good luck with your next projects! :rocket:
#### Happy Coding! :+1:

__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Pull request merged**